dev-ruby/facter/facter-3.11.3-r1.ebuild

```diff
@@ -40,23 +38,29 @@ CDEPEND="
 
 ruby_add_bdepend "test? ( dev-ruby/rake dev-ruby/rspec:2 dev-ruby/mocha:0.14 )"
 
-RDEPEND="${CDEPEND}"
+RDEPEND="${COMMON_DEPEND}"
 DEPEND="${BDEPEND}
-	${CDEPEND}"
+	${COMMON_DEPEND}"
+
+# restore ${S} and override all phases exported by ruby-ng.eclass
+S="${WORKDIR}/${P}"
+
+pkg_setup() {
```

Comment:
ruby isn't actually needed for this package to work, so no need to run it's pkg_setup

Reply:
This package already depends on ruby via ruby-ng.eclass (RUBY_OPTIONAL != yes). It can be made optional, but this should be done via use flag. Otherwise we can't properly install facter.rb for all ruby targets.

Reply:
ya, should probably be made optional then

Reply:
Don't expect me to do that in this PR as it probably require changes in reverse deps (app-admin/puppet). :)

Reply:
ok, but you can do multiple commits in a PR, but that's up to you :P

Reply:
Unfortunately, build system does not allow to disable ruby support. So this needs either additional hacks in the ebuild or patches for build system.